src/components/Schedule/Schedule.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Card } from '../Common';
+import { EVENT_WIDTH } from './Constants'
+import { TimelineColumn } from './Timeline'
+import { TagLegend } from './Tag'
+import Event from './Event'
+
+const ScheduleFlexContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`
+
+const FlexColumn = styled.div`
+  flex: 0 0 ${EVENT_WIDTH}px;
+`
+
+const msToHours = ms => ms / 1000 / 60 / 60
+
+const ScheduleColumn = ({ column }) => {
+  return (
+    <FlexColumn>
+      {column.map((event) => <Event event={event} />)}
+    </FlexColumn>
+  )
+}
+
+export default ({ events, hackathonStart, hackathonEnd }) => {
+  const produceOptimalSchedule = (events) => {
+    const res = []
+    let unusedEvents = events;
+
+    // build optimal columns list while there are still events that
+    // are not in the schedule
+    while (unusedEvents.length > 0) {
+      let latestTime = new Date(0) // epoch time lmao
+      let usedIndices = []
+
+      // produce optimal schedule and track which events are used
+      const sched = unusedEvents.reduce((accumulator, event, i) => {
+        const curTime = new Date(event.startTime)
+        if (curTime.getTime() >= latestTime.getTime()) {
+          // set offset as number of hours between start of hackathon and this event
+          const hoursFromStart = msToHours(curTime - hackathonStart)
+          event.hourOffset = hoursFromStart
+
+          // set duration of event
+          const duration = msToHours((new Date(event.endTime)) - (new Date(event.startTime)))
+          event.duration = duration
+
+          accumulator.push(event)
+          usedIndices.push(i)
+          latestTime = new Date(event.endTime)
+        }
+        return accumulator;
+      }, [])
+
+      // update unused by removing used events
+      unusedEvents = unusedEvents.filter((value, index) => {
+        return usedIndices.indexOf(index) === -1;
+      })
```

Comment:
I'm guessing theres no way to remove from `unusedEvents` while you are iterating thru it the first time.

Also maybe we can use `.includes` instead of `indexOf` to make the code slightly more intuitive?

Reply:
yeah couldn't figure out how to modify the thing you're reducing on lmao

will include that change though